test/unit/master/test_pagination.py

```diff
@@ -0,0 +1,118 @@
+from typing import Optional
+from threading import Event
+
+from genty import genty, genty_dataset
+from hypothesis import given
+from hypothesis.strategies import text, dictionaries, integers
+from unittest.mock import MagicMock, Mock
+
+from app.master.build import Build
+from app.master.build_request import BuildRequest
+from app.master.cluster_master import ClusterMaster
+from app.slave.cluster_slave import SlaveState
+from app.util.exceptions import BadRequestError, ItemNotFoundError
+from app.web_framework.cluster_base_handler import ClusterBaseHandler, pagination_constants
+from test.framework.base_unit_test_case import BaseUnitTestCase
+
+
+@genty
+class TestClusterMaster(BaseUnitTestCase):
```

Comment:
The test class name should match the name of the file. And on that note, why a new file instead of adding these to the existing test_cluster_master.py? (Edit: see my comment below; I think I understand why you might have split these out.)